### DISCUSSION TITLE
Question about __bbox_clip(self, cx, cy, width, height, boundary) in siamrpn_tracker.py

### DISCUSSION THREAD

**Discussion by ArronYao** [STVIR/pysot#364] [General] [answered]
Could you explain the  function _bbox_clip(self, cx, cy, width, height, boundary)  in more detail?
I think this function is quite confusing. What roles do cx and cy play here? Are they mean centor_x and centor_y respectively? If so, why do you calculate cx and cy by the following codes (line157 ,158 in siamrpn_tracker.py)?
        cx = bbox[0] + self.center_pos[0]
        cy = bbox[1] + self.center_pos[1]

This question really confuses me for a long time. I will be very appreciated that if you could do some explanation for this question.

**Comment by ZhiyuanChen** [comment, marked as answer]
If I recall right, bbox here is cxywh format. bbox[0] and bbox[1] represent the offset of center x and y respectfully.
